File: fixtures/parser/basic-constructs/shapes.ts
function readonly(target: unknown, key: string): void {
  void target;
  void key;
}

export interface Point {
  x: number;
  y: number;
}

export type Coordinates = [number, number];

export interface Sized {
  area(): number;
}

export enum Direction {
  Up,
  Down,
  Left,
  Right,
}

export abstract class Shape {
  @readonly
  readonly id: string;

  protected color: string = "black";

  constructor(id: string) {
    this.id = id;
  }

  abstract area(): number;

  describe(): string {
    return `${this.id} is ${this.color}`;
  }
}

export class Circle extends Shape implements Point, Sized {
  x = 0;
  y = 0;

  static defaultColor: string = "black";

  constructor(
    id: string,
    private radius: number,
  ) {
    super(id);
  }

  area(): number {
    return Math.PI * this.radius * this.radius;
  }

  get diameter(): number {
    return this.radius * 2;
  }

  set diameter(value: number) {
    this.radius = value / 2;
  }
}

class InternalHelper {
  help(): string {
    return "help";
  }
}

File: fixtures/parser/python-constructs/shapes.py
class Sized:
    def area(self):
        pass


class Shape:
    def __init__(self, name):
        self.name = name

    def describe(self):
        return self.name


@dataclass
class Circle(Shape, Sized):
    def __init__(self, name, radius):
        super().__init__(name)
        self.radius = radius

    def area(self):
        return 3.14159 * self.radius * self.radius


class _InternalHelper:
    def help(self):
        return "help"
